### PR TITLE
tests: add namespace route

### PR DESCRIPTION
**Pull request by codingben**
- Added namespace route tests.
- Added setup unit tests.
- Updated path parameters.
- Updated model types.

### CODE REVIEW

File: src/controllers/namespace.go [hunk 1]
```diff
@@ -19,12 +19,12 @@ type NamespaceController interface {
 }
 
 type namespaceController struct {
-	client *kubernetes.Clientset
+	client kubernetes.Interface
```
Comment (mzeevi) [inline src/controllers/namespace.go:22]: Why is this change needed?

File: cmd/main.go [hunk 2]
```diff
@@ -41,12 +41,15 @@ func main() {
...
-		namespacesGroup.POST("/", routes.CreateNamespace(client, logger))
-		namespacesGroup.DELETE("/:name", routes.DeleteNamespace(client, logger))
+		namespacesGroup := v1.Group("/namespaces")
+		{
+			namespacesGroup.GET("/", routes.ListNamespaces(client, logger))
+			namespacesGroup.GET("/:name", routes.GetNamespace(client, logger))
+			namespacesGroup.POST("/", routes.CreateNamespace(client, logger))
+			namespacesGroup.DELETE("/:name", routes.DeleteNamespace(client, logger))
+		}
 	}
```
Comment (mzeevi) [inline cmd/main.go:53]: This change exists in another PR as well, no?

**Review by mzeevi — COMMENTED** [review]
Overall looks fine, I'll leave LGTM-ing to @fran2810 

We should decide whether we want to go with Ginkgo or just the default Testing Framework (or both)

File: src/controllers/namespace.go (hunk 1 above)
Comment (codingben) [reply src/controllers/namespace.go:22]: It's needed in order to use `fake.Clientset` which implements `clientset.Interface` and not `kubernetes.Clientset`. See here:
```
k8s.io/client-go/kubernetes/fake/clientset_generated.go

// Clientset implements clientset.Interface. Meant to be embedded into a
// struct to get a default implementation. This makes faking out just the method
// you want to test easier.
type Clientset struct {
	testing.Fake
	discovery *fakediscovery.FakeDiscovery
	tracker   testing.ObjectTracker
}
```

File: cmd/main.go (hunk 2 above)
Comment (codingben) [reply cmd/main.go:53]: Yes, I need here it in the tests so I can write tests to execute http requests on `/v1/...` path, it's needed so we won't send another PR to refactor it later.

**Review by mzeevi — CHANGES_REQUESTED** [review]

**Review by mzeevi — APPROVED** [review]
/lgtm
/approve
